### ISSUE TITLE
[YSQL] Core dump on catalog version mismatch then commit

### ISSUE THREAD

**Issue by jaki** [yugabyte/yugabyte-db#5196]
Jira Link: [DB-1531](https://yugabyte.atlassian.net/browse/DB-1531)
```sh
./bin/yb-ctl create --rf 3
```

Then, create two `ysqlsh` session on the same host, and execute commands as follows.

Session A | Session B |
| --- | --- |
| `CREATE TABLE t (i int)` | |
| `BEGIN ISOLATION LEVEL SERIALIZABLE` | |
| | `BEGIN ISOLATION LEVEL SERIALIZABLE` |
| `SELECT 1` | |
| | `SELECT 1` |
| `ALTER TABLE t ADD COLUMN a int` | |
| | `INSERT INTO t (a) VALUES (1) -- quickly` |
| `ALTER TABLE t ADD COLUMN b int` | |
| | `INSERT INTO t (b) VALUES (1) -- quickly` |
| ... | ... |
| | `COMMIT` |

Keep doing `ALTER TABLE` then (immediately) `INSERT` until you get a catalog version mismatch.  Then, `COMMIT`, and that causes a core dump for the `postgres` process.

Backtrace of the core:

```sh
gdb build/latest/postgres/bin/postgres /path/to/yugabyte-data/node-1/disk-1/pg_data/core.xxxxx -ex bt
```

```
#0  0x00007f3cccb750a7 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007f3cccb764aa in __GI_abort () at abort.c:89
#2  0x000000000099f571 in ExceptionalCondition (conditionName=conditionName@entry=0xb735fc "!(IsTransactionState())", errorType=errorType@entry=0x9fbb80 "FailedAssertion", fileName=fileName@entry=0xbaddd0 "../../../../../../../src/postgres/src/backend/utils/cache/catcache.c", lineNumber=lineNumber@entry=1564) at ../../../../../../../src/postgres/src/backend/utils/error/assert.c:54
#3  0x00000000009823cc in SearchCatCacheInternal (v4=0, v3=0, v2=0, v1=12457, nkeys=1, cache=0x1c54180) at ../../../../../../../src/postgres/src/backend/utils/cache/catcache.c:1564
#4  SearchCatCache1 (cache=0x1c54180, v1=12457) at ../../../../../../../src/postgres/src/backend/utils/cache/catcache.c:1518
#5  0x000000000099a228 in SearchSysCache1 (cacheId=cacheId@entry=21, key1=<optimized out>) at ../../../../../../../src/postgres/src/backend/utils/cache/syscache.c:1466
#6  0x00000000006287af in get_database_name (dbid=<optimized out>) at ../../../../../../src/postgres/src/backend/commands/dbcommands.c:2209
#7  0x000000000099051c in YBPreloadRelCache () at ../../../../../../../src/postgres/src/backend/utils/cache/relcache.c:1294
#8  0x000000000085d417 in YBRefreshCache () at ../../../../../../src/postgres/src/backend/tcop/postgres.c:3707
#9  0x00000000008632f2 in YBCheckSharedCatalogCacheVersion () at ../../../../../../src/postgres/src/backend/tcop/postgres.c:3950
#10 PostgresMain (argc=<optimized out>, argv=argv@entry=0x1be9680, dbname=0x1be94e0 "yugabyte", username=0x1be94b8 "yugabyte") at ../../../../../../src/postgres/src/backend/tcop/postgres.c:4830
#11 0x00000000007cfc12 in BackendRun (port=0x1c13f80) at ../../../../../../src/postgres/src/backend/postmaster/postmaster.c:4428
#12 BackendStartup (port=0x1c13f80) at ../../../../../../src/postgres/src/backend/postmaster/postmaster.c:4094
#13 ServerLoop () at ../../../../../../src/postgres/src/backend/postmaster/postmaster.c:1737
#14 PostmasterMain (argc=argc@entry=23, argv=argv@entry=0x1be7bc0) at ../../../../../../src/postgres/src/backend/postmaster/postmaster.c:1400
#15 0x0000000000726290 in PostgresServerProcessMain (argc=23, argv=0x1be7bc0) at ../../../../../../src/postgres/src/backend/main/main.c:234
#16 0x000000000072632c in main ()
```

This was on commit 6d8b235f988ffb5071a675ac98eec1507bccce32, centos debug build.  I wasn't able to reproduce on `--rf 1`.

**Comment by kriteshranjan** [comment]
AI validation notes (`yb-issue-5196-analysis.md`):

# Analysis: yugabyte/yugabyte-db#5196 (DB-1531)

Issue: "[YSQL] Core dump on catalog version mismatch then commit" (opened 2020-07-23 by a YB contributor, still open, 0 comments, labels area/ysql kind/bug priority/medium). Reported on commit `6d8b235f988ffb5071a675ac98eec1507bccce32`, centos DEBUG (assert-enabled) build, `yb-ctl create --rf 3`.
Signature: `ExceptionalCondition("!(IsTransactionState())", FailedAssertion, catcache.c:1564)` <- `SearchCatCacheInternal` <- `SearchSysCache1` <- `get_database_name` (dbcommands.c:2209) <- `YBPreloadRelCache` (relcache.c:1294) <- `YBRefreshCache` (postgres.c:3707) <- `YBCheckSharedCatalogCacheVersion` (postgres.c:3950) <- `PostgresMain` (postgres.c:4830).
Kind: runtime crash (debug-build assertion; on a release build the same path runs catalog reads in `TRANS_ABORT` state without the assert).

## Verdict
NOT-REPRODUCED — the reported crash path is fully explained in the 2020 source and was closed by `3253827a31` (2020-10-08, first tag v2.3.3.0, first stable 2.4.0); all five current in-support targets carry the fix and survive the reporter's scenario at runtime (5 images x 6 cells, 0 crash signatures, 0 client disconnects). No assert-enabled build of the 2020 revision exists as an image, so the assertion itself was not re-triggered. Quick disposition: Fixed; no backports needed.

## Root cause
2020 code at `6d8b235f98`:
- `src/postgres/src/backend/tcop/postgres.c:3931-3952` `YBCheckSharedCatalogCacheVersion()` gated the refresh on the per-statement static `xact_started` (`:3936`), not on the transaction *block* state. `xact_started` is cleared to `false` in PostgresMain's error-recovery block (`:4667`) even when the transaction block stays open in `TBLOCK_ABORT`.
- Scenario: session B is in `BEGIN ISOLATION LEVEL SERIALIZABLE`; its `INSERT` fails with the in-txn-block re-throw `ERROR: Catalog Version Mismatch: A DDL occurred while processing this query. Try Again.` (`postgres.c:3843-3847`, raised because `consider_retry && !IsTransactionBlock()` at `:3803` is false inside a block). PG error recovery runs `AbortCurrentTransaction()` -> block state `TBLOCK_ABORT`, `xact_started = false`.
- B then sends `COMMIT`. Before dispatching the command, `PostgresMain` calls `YBCheckSharedCatalogCacheVersion()` (`postgres.c:4830`). `xact_started` is false so the guard passes; the shared catalog version (bumped by session A's `ALTER TABLE`) exceeds `yb_catalog_cache_version`, so `YBRefreshCache()` runs (`:3950`). Its own guard is also `xact_started` (`:3690`), so it passes, and it calls `start_xact_command()` (`:3699`) -> `StartTransactionCommand()`, which is a **no-op in `TBLOCK_ABORT`** (`src/postgres/src/backend/access/transam/xact.c:2795` + the `case TBLOCK_ABORT: case TBLOCK_SUBABORT: break;` arm). The transaction state remains `TRANS_ABORT`.
- `YBPreloadRelCache()` (`relcache.c:1281`, `get_database_name(MyDatabaseId)` at `:1294`) -> `SearchSysCache1` -> `SearchCatCacheInternal` hits `Assert(IsTransactionState())` (`src/postgres/src/backend/utils/cache/catcache.c:1564`; `IsTransactionState()` is `state == TRANS_INPROGRESS`, `xact.c:357-368`) -> abort() -> core. Any error inside the block (not only the mismatch) followed by a catalog-version bump before the next command reproduces the same path; the mismatch was just the reporter's trigger. RF3 vs RF1 only changed the odds of the version bump landing in the window.

Fix (`3253827a315ac2ea5dc336ade8c76172fec5d9e3`, "[YSQL] #3979 #4360 Reduce version mismatch errors and improve fault tolerance for DDLs"): `if (xact_started)` -> `if (IsTransactionOrTransactionBlock())` in `YBCheckSharedCatalogCacheVersion` (postgres.c:3936-3937 at that commit). `IsTransactionOrTransactionBlock()` is true for any `blockState != TBLOCK_DEFAULT`, including `TBLOCK_ABORT` (`xact.c:5415-5423` on master), so no refresh is attempted while a block is open or aborted; the refresh happens on the first command after `COMMIT`/`ROLLBACK` returns the block to `TBLOCK_DEFAULT`.

Current code (same shape on every resolved target; guard is the second statement of the function):
| revision | `YBCheckSharedCatalogCacheVersion` | guard `if (IsTransactionOrTransactionBlock()) return;` | `YBRefreshCache` | call site in `PostgresMain` |
| master `6c3faf42b7` (= local `d666a274f0`, function byte-identical) | postgres.c:5222 | :5228 | :4692 (still keeps `if (xact_started)` at :4704 as an inner guard) | :7045 |
| v2026.1.1.1 `3b010e021f` | :5220 | :5226 | :4690 | :6967 |
| v2025.2.6.0 `ae22ef35ff` | :5230 | :5236 | :4687 | :6995 |
| v2025.1.4.0 `690ac4141e` | :4886 | :4892 | :4404 | :6563 |
| v2024.2.11.0 `04544fd94f` | :4293 | :4298 | :3931 | :5800 |
| v2.25.2.0 `03bbc9f31e` | :4749 | :4755 | :4299 | :6340 |
The ready-for-query path is likewise guarded (`IsAbortedTransactionBlockState()` / `IsTransactionOrTransactionBlock()` before the `yb_need_cache_refresh` refresh, master postgres.c:6814-6845), and the retry path refreshes only when `!IsTransactionBlock()` (`YBPrepareCacheRefreshIfNeeded`). No remaining caller can reach `YBRefreshCache` with an aborted block. Affected in-support targets: none.

## Trigger matrix
Predictions (stated before running; all five images carry the fix): B's COMMIT/ROLLBACK after an in-txn error + concurrent catalog bump returns `ROLLBACK`, backend survives, the next statement works and performs the deferred refresh; controls commit normally.

| cell | condition | predicted | observed (all 5 targets) |
| R1 | both sessions SERIALIZABLE; A `ALTER TABLE t ADD COLUMN cN`, B immediately `INSERT INTO t (cN)` until error; B `COMMIT`; B `SELECT` | B errors (modern: 42703 stale relcache, ADD COLUMN is non-breaking `pg_yb_utils.c:4112-4119`), COMMIT -> `ROLLBACK`, alive | n=1 ok, n=2 `ERROR: 42703: column "c2" of relation "t" does not exist`; `COMMIT` -> `ROLLBACK`; B alive; refresh logged on next stmt |
| R2 | as R1 but B `ROLLBACK` | `ROLLBACK`, alive | 42703 at n=1 (relcache already cached), `ROLLBACK` -> `ROLLBACK`, alive |
| R3 | B SERIALIZABLE txn reads t; A breaking DDL (`RENAME COLUMN` there-and-back); B `INSERT INTO t (i)`; B `COMMIT` | B gets 40001 (modern equivalent of the reported mismatch), COMMIT -> `ROLLBACK`, alive | `ERROR: 40001: current transaction is expired or aborted ...` (2.25.2: `could not serialize access due to concurrent update`); `COMMIT` -> `ROLLBACK`; alive |
| R4 | deterministic: B `SELECT 1/0` (aborted block); A `ADD COLUMN` + `RENAME` (version bump); sleep 3 s; B `COMMIT`; B `SELECT` | exact 2020 crash precondition (aborted block + shared > local); `ROLLBACK`, alive | `COMMIT` -> `ROLLBACK`; alive; `YBCheckSharedCatalogCacheVersion: got tserver catalog version: N` logged for B only at the follow-up SELECT |
| C1 | control: B txn `INSERT`, `COMMIT`, no DDL | `COMMIT` | `COMMIT`, alive |
| C2 | control: B `SELECT 1/0`, `COMMIT`, no DDL | `ROLLBACK`, alive | `ROLLBACK`, alive |
| log scan | `terminated by signal|SIGSEGV|SIGABRT|TRAP:|PANIC|was terminated|server process .* exited|IsTransactionState` in postgresql-*.log + yb-tserver.INFO; `server closed the connection unexpectedly` in transcripts | 0 / 0 | 0 / 0 on every target (one startup-time `server process (PID 286) exited with exit code 1` WARNING in one 2026.1 run, 3 s before A/B connected, exit code not signal; 0 in the other 2026.1 run) |

Adjustments vs the 2020 report (documented, asserted behavior unchanged): RF1 single-node yugabyted instead of `yb-ctl --rf 3` (the version bump is delivered deterministically by heartbeat; R4 removes the race entirely); the literal "Catalog Version Mismatch" wording is unreachable on modern builds because `ALTER TABLE ADD COLUMN` is non-breaking and statements inside a block never refresh, so R1/R2 fail with 42703 and R3 supplies the breaking-DDL 40001 equivalent.

## Reproduction
Bundle: `/tmp/claude/yb5196-reproduce.sh.txt` (host runner `run_target.sh` + in-container `driver.sh`, two long-lived `ysqlsh` sessions over FIFOs with `\echo` markers, `\set VERBOSITY verbose`). Evidence: `/tmp/claude/yb5196-matrix.txt` (per-target RESULT lines, crash-scan matches, B-backend catalog-version log lines, COMMIT/ROLLBACK answer counts); raw per-target transcripts `/tmp/claude/yb5196/matrix-<tag>.txt`.

```sh
# host (podman, arm64); one target at a time
export REGISTRY_AUTH_FILE=/tmp/claude/yb5196/auth/config.json DOCKER_CONFIG=/tmp/claude/yb5196/auth   # dir holding "{}"
bash /tmp/claude/yb5196/run_target.sh 2026.1.1.1-b2       # also: 2025.2.6.0-b111 2024.2.11.0-b36
YB5196_FLAGS='reject_writes_when_disk_full=false,log_ysql_catalog_versions=true' bash /tmp/claude/yb5196/run_target.sh 2025.1.4.0-b103   # and 2.25.2.0-b359
# runner: podman run -d --name yb5196-<tag> --memory 2400m --entrypoint /bin/sleep docker.io/yugabytedb/yugabyte:<tag>-aarch64 infinity
#         podman exec ... /home/yugabyte/bin/yugabyted start --advertise_address 127.0.0.1 --base_dir /tmp/ybd --master_flags "$FLAGS" --tserver_flags "$FLAGS"
#         podman exec -i ... bash -s < driver.sh ; teardown: yugabyted stop; podman rm -f
```
Driver core (per session, `ysqlsh -h 127.0.0.1 -p 5433 -U yugabyte -X -a`): A: `CREATE TABLE t (i int)`; both: `BEGIN ISOLATION LEVEL SERIALIZABLE; SELECT 1;`; loop A `ALTER TABLE t ADD COLUMN cN int;` -> B `INSERT INTO t (cN) VALUES (1);` until B errors; B `COMMIT;` then `SELECT 'ALIVE_' || pg_backend_pid();`. Success signature (fixed): `COMMIT` answers `ROLLBACK`, ALIVE row returned, postgres log shows `YBCheckSharedCatalogCacheVersion: got tserver catalog version: N` for B's PID only after the COMMIT (flag `log_ysql_catalog_versions=true`), crash-scan count 0. Failure signature (2020 debug build): backend abort with `TRAP: FailedAssertion("!(IsTransactionState())", File: ".../catcache.c", Line: 1564)` / `server closed the connection unexpectedly` on COMMIT.
Flag adjustment: `reject_writes_min_disk_space_pct` does not exist on 2025.1.4.0-b103 / 2.25.2.0-b359 (yb-master exits `ERROR: unknown command line flag`, `--undefok` did not rescue it), so those two ran without it; all targets ran with `reject_writes_when_disk_full=false,log_ysql_catalog_versions=true`.

## Versions tested
Resolved at: 2026-09-10T15:14:46Z–15:17Z UTC (controller resolution; re-verified locally 2026-09-10T15:24Z: all five tag SHAs present and matching `git rev-parse`, images present by digest)
Resolution evidence: Docker Hub tags `?name=2024.2.&ordering=last_updated` newest 2024.2.11.0-b36 (87 tags; pushed 2026-08-25T16:02:12Z), arm64 digest sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288; `?name=2025.1.` newest 2025.1.4.0-b103 (36; 2026-04-20T15:59:42Z), sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0; `?name=2025.2.` newest 2025.2.6.0-b111 (39; 2026-09-04T22:23:02Z), sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e; `?name=2026.1.` newest 2026.1.1.1-b2 (= `latest`, digest sha256:3926eedf0ff4b99426fbff533128ff16635ed5697a5cf086dde4fafec28e6f75) (12; 2026-08-18T20:03:16Z), arm64 sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699; master preview `?name=2.25.` newest 2.25.2.0-b359 (9; 2025-05-20T18:34:42Z), `?name=2.27.`/`?name=2.29.` count 0, arm64 sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5. Source tags: v2024.2.11.0 -> 04544fd94fb4e50d05994784f444b075a4a5dfdb, v2025.1.4.0 -> 690ac4141e57d060fd834312f396643e8c41f260, v2025.2.6.0 -> ae22ef35ff65de63303467d95ac3e6d2382feba9, v2026.1.1.1 -> 3b010e021f6027744bf9e016b17e108a2eb24e44, v2.25.2.0 -> 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d. Controller master HEAD at resolution: 6c3faf42b719960511e288d2e533f858070a7e4e (committed 2026-09-10T14:54:24Z). Observed master HEAD (local `origin/master`, shallow clone, checked 2026-09-10T15:22Z): d666a274f0fe6e3fadbce2e3c845f5070afa8d66 (2026-09-01); master census done on 6c3faf42b7 via GitHub raw contents, function identical to d666a274f0. Image digests observed via `podman image inspect` matched the map for all five images.

| target | latest resolved version | source revision | observed repro/control | effective status |
| 2024.2 | 2024.2.11.0-b36 (`PostgreSQL 11.2-YB-2024.2.11.0-b0`) | 04544fd94fb4e50d05994784f444b075a4a5dfdb | R1/R2 42703 then `ROLLBACK`, alive; R3 40001 then `ROLLBACK`, alive; R4 `ROLLBACK`, alive; C1 `COMMIT`; C2 `ROLLBACK`; crash scan 0; disconnects 0; B refresh logged after COMMIT (`Refreshing catalog cache.` x3) | fixed |
| 2025.1 | 2025.1.4.0-b103 (`PostgreSQL 15.12-YB-2025.1.4.0-b0`) | 690ac4141e57d060fd834312f396643e8c41f260 | same six outcomes; crash scan 0; disconnects 0; B refresh events 3 (versions 4,6,8) | fixed |
| 2025.2 | 2025.2.6.0-b111 (`PostgreSQL 15.12-YB-2025.2.6.0-b0`) | ae22ef35ff65de63303467d95ac3e6d2382feba9 | same six outcomes; crash scan 0; disconnects 0; B refresh events 2 (6,8) | fixed |
| 2026.1 | 2026.1.1.1-b2 (`PostgreSQL 15.12-YB-2026.1.1.1-b0`) | 3b010e021f6027744bf9e016b17e108a2eb24e44 | same six outcomes in both runs; crash scan 0 (run 1) / 1 startup-time exit-code-1 WARNING before sessions existed (run 2); disconnects 0; B refresh events 4 (2,5,7,9) | fixed |
| master | preview 2.25.2.0-b359 (`PostgreSQL 15.12-YB-2.25.2.0-b0`) | 03bbc9f31eae5c3fecabd8f0afa5e99211e6080d (master HEAD 6c3faf42b7 source-censused: guard postgres.c:5228) | same six outcomes (R3 text `could not serialize access due to concurrent update`); crash scan 0; disconnects 0; B refresh events 2 (6,8) | fixed |
| reported (source-only, no image) | 2020-07-21 master build | 6d8b235f988ffb5071a675ac98eec1507bccce32 | not executed; mechanism confirmed by reading postgres.c:3936/3950/3690/3699/4667/4830, xact.c:2795 TBLOCK_ABORT arm, catcache.c:1564 | affected (historical) |

First-fixed provenance (tag/commit ancestry, separate from the runtime proof above): fix `3253827a31` (2020-10-08T03:22:27Z). GitHub compare `3253827a31...v2.3.2.0` = behind 107 (not contained; v2.3.2.0 postgres.c:3934 still `if (xact_started)`); `3253827a31...v2.3.3.0` = ahead 111 (contained; v2.3.3.0 postgres.c:3934 `if (IsTransactionOrTransactionBlock())`). First release: v2.3.3.0 (2020-10-24, preview line). First stable: v2.4.0 (2021-01-21). The 2.2 stable line never received it (v2.2.4 ... v2.2.7 of 2021-02-02 still carry `if (xact_started)`). Every in-support branch (2024.2, 2025.1, 2025.2, 2026.1, master) descends from 2.4+ and contains the guard (census table above). Local clone is shallow (23 grafts), so provenance used GitHub contents/compare APIs. No commit references #5196 (the `git log --grep 5196` hits are Phorge `D5196x` numbers).

## Proposed fix direction
No product fix needed; recommend closing #5196 as fixed by `3253827a31` (#3979/#4360). Optional hardening (master only, not required): replace the inner `if (xact_started)` check in `YBRefreshCache` (postgres.c:4704) with `Assert(!IsTransactionOrTransactionBlock())` plus the existing ereport, so any future caller that reintroduces an in-block refresh fails loudly in debug builds rather than reading catalogs in `TRANS_ABORT`. Alternative considered: nothing (all three callers are already guarded); the assert is cheap insurance only if a maintainer wants it.

## Test strategy
Existing coverage: `java/yb-pgsql/src/test/java/org/yb/pgsql/TestPgDdlFaultTolerance.java` (added by the fix commit), `TestPgTransactions.java`, `src/yb/yql/pgwrapper/pg_catalog_version-test.cc`, `pg_ddl_concurrency-test.cc`, `pg_ddl_atomicity-test.cc`. A targeted regression via public entry points: two libpq connections, conn B `BEGIN ISOLATION LEVEL SERIALIZABLE; SELECT 1; SELECT 1/0;` (aborted block), conn A `ALTER TABLE t ADD COLUMN x int; ALTER TABLE t RENAME COLUMN x TO y;` (catalog version bump), wait for `yb_catalog_version()`/heartbeat (>= 2 s or poll `pg_stat_activity` catalog version), conn B `COMMIT` (expect `ROLLBACK` status, connection still `CONNECTION_OK`), then `SELECT 1`. Runs in a debug (`Assert`-enabled) build, e.g. `./yb_build.sh debug --cxx-test pgwrapper_pg_catalog_version-test --gtest_filter 'PgCatalogVersionTest.CommitAfterErrorWithConcurrentDdl'`; the pre-fix code would trip `catcache.c` `Assert(IsTransactionState())`.

## Confidentiality check
Artifact, evidence files, and scripts contain only public issue content, public commit SHAs/tags, public Docker Hub digests, synthetic table `t`, loopback addresses, and local `/tmp/claude` paths. No customer data, PII, credentials, or internal hostnames/URLs. The `auth/config.json` used for podman is the literal `{}`.

## Side findings
- Modern builds cannot emit the reporter's literal "Catalog Version Mismatch" for this shape: `ALTER TABLE` is non-breaking (`pg_yb_utils.c:4112-4119`, "We rely on table schema version mismatch to abort transactions that touch the table") and statements inside a block do not refresh, so B's stale relcache yields 42703; breaking DDL (RENAME) instead aborts the in-flight SERIALIZABLE txn with 40001 (`current transaction is expired or aborted` / `could not serialize access`).
- `YBRefreshCache` still uses the per-statement `xact_started` as its inner guard on master (postgres.c:4704); correctness rests entirely on the callers' `IsTransactionOrTransactionBlock()` checks.
- Runtime gotcha: `--undefok=reject_writes_min_disk_space_pct` does not suppress the unknown-flag failure on 2025.1.4.0-b103 / 2.25.2.0-b359 yb-master (`master.err`: `ERROR: unknown command line flag 'reject_writes_min_disk_space_pct'`); 2024.2.11.0-b36, 2025.2.6.0-b111 and 2026.1.1.1-b2 accept the flag.

**Comment by kriteshranjan** [comment]
AI validation notes (`yb5196-reproduce.sh.txt`):

# yb5196 reproduction bundle (public-safe). Two files concatenated: run_target.sh (host) and driver.sh (in-container).
# Usage on host with podman: bash run_target.sh <image-tag>   e.g. 2026.1.1.1-b2  (images docker.io/yugabytedb/yugabyte:<tag>-aarch64)
# For 2025.1.4.0-b103 and 2.25.2.0-b359 set YB5196_FLAGS="reject_writes_when_disk_full=false,log_ysql_catalog_versions=true" (reject_writes_min_disk_space_pct does not exist there).

######## run_target.sh ########
#!/bin/bash
# yb5196 per-target runner (host side). Usage: run_target.sh <image-tag> [rf3]
# Starts one container with a single-node yugabyted (RF1), runs driver.sh inside it, saves output, tears down.
set -u
TAG=$1; MODE=${2:-rf1}
export REGISTRY_AUTH_FILE=/tmp/claude/yb5196/auth/config.json DOCKER_CONFIG=/tmp/claude/yb5196/auth
IMG="docker.io/yugabytedb/yugabyte:${TAG}-aarch64"
NAME="yb5196-$(printf '%s' "$TAG" | tr '.' '-')"
OUT="/tmp/claude/yb5196/matrix-${TAG}.txt"
FLAGS="${YB5196_FLAGS:-reject_writes_when_disk_full=false,reject_writes_min_disk_space_pct=0,undefok=reject_writes_min_disk_space_pct,log_ysql_catalog_versions=true}"
podman rm -f "$NAME" >/dev/null 2>&1
{
echo "===== yb5196 target=$TAG image=$IMG container=$NAME mode=$MODE ====="
echo "image digest: $(podman image inspect "$IMG" --format '{{.Digest}}')"
podman run -d --name "$NAME" --memory 2400m --entrypoint /bin/sleep "$IMG" infinity >/dev/null || { echo "RUN FAILED"; exit 1; }
podman exec "$NAME" bash -c "/home/yugabyte/bin/yugabyted start --advertise_address 127.0.0.1 --base_dir /tmp/ybd --master_flags '$FLAGS' --tserver_flags '$FLAGS' 2>&1 | tail -3"
# wait for YSQL
for i in $(seq 1 90); do
  podman exec "$NAME" /home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte -Atc 'SELECT 1' >/dev/null 2>&1 && break
  sleep 2
done
podman exec "$NAME" /home/yugabyte/bin/ysqlsh -h 127.0.0.1 -U yugabyte -d yugabyte -Atc "SELECT version();" 2>&1
podman exec "$NAME" bash -c "/home/yugabyte/bin/yugabyted status --base_dir /tmp/ybd 2>&1 | grep -E 'Status|Replication' | head -3"
echo "----- driver -----"
podman exec -i "$NAME" bash -s < /tmp/claude/yb5196/driver.sh 2>&1
echo "----- postmaster/backend crash scan (postgresql-*.log + yb-tserver.INFO) -----"
podman exec "$NAME" bash -c 'grep -H -n -E "terminated by signal|SIGSEGV|SIGABRT|TRAP:|PANIC|IsTransactionState|server process .* (exited|was terminated)" /tmp/ybd/data/yb-data/tserver/logs/postgresql-*.log /tmp/ybd/data/yb-data/tserver/logs/yb-tserver.INFO 2>/dev/null | head -20; echo "scan_exit=$?"'
podman exec "$NAME" bash -c 'ls -la /tmp/ybd/data/yb-data/tserver/logs/ | grep -i -E "core|postgresql" | head; ls /tmp/ybd/data/pg_data/core* 2>/dev/null | head -3; echo "cores_listed_above"'
echo "----- teardown -----"
podman exec "$NAME" bash -c "/home/yugabyte/bin/yugabyted stop --base_dir /tmp/ybd >/dev/null 2>&1; echo stopped"
podman rm -f "$NAME" >/dev/null 2>&1 && echo "removed $NAME"
echo "===== end $TAG ====="
} 2>&1 | tee "$OUT"

######## driver.sh ########
#!/bin/bash
# yb5196 driver — runs INSIDE the yugabyte container (fed via `podman exec -i ... bash -s`).
# Two long-lived ysqlsh sessions (A = DDL, B = victim) driven through FIFOs.
# Cells:
#   R1  reporter literal: both BEGIN ISOLATION LEVEL SERIALIZABLE; SELECT 1; loop A: ALTER TABLE t ADD COLUMN cN int,
#       B immediately INSERT INTO t (cN) VALUES (1) until B errors; then B COMMIT; then B SELECT 1.
#   R2  same as R1 but B ROLLBACK instead of COMMIT; then B SELECT 1.
#   R3  modern mismatch variant: B in SERIALIZABLE txn touches t; A runs a BREAKING DDL (RENAME COLUMN there-and-back);
#       B INSERT INTO t (i) -> expect catalog-version-mismatch class error (40001); B COMMIT; B SELECT 1.
#   R4  deterministic aborted-block + version bump: B SELECT 1/0 (aborted), A ALTER ADD COLUMN (bump), sleep 3s
#       (heartbeat), B COMMIT; B SELECT 1.  (exercises the exact 2020 crash path without a race)
#   C1  control: B SERIALIZABLE txn, INSERT, COMMIT, no concurrent DDL.
#   C2  control: B SERIALIZABLE txn, SELECT 1/0, COMMIT, no DDL.
set -u
BIN=/home/yugabyte/bin
W=/tmp/yb5196
rm -rf "$W"; mkdir -p "$W"
PSQL="$BIN/ysqlsh -h 127.0.0.1 -p 5433 -U yugabyte -d yugabyte -X -a"
LOG="$W/driver.log"
log() { printf "%s %s\n" "$(awk "BEGIN{srand(); print srand()}")" "$*" | tee -a "$LOG" >&2; }

mkfifo "$W/a.in" "$W/b.in"
$PSQL < "$W/a.in" > "$W/a.out" 2>&1 &
PA=$!
$PSQL < "$W/b.in" > "$W/b.out" 2>&1 &
PB=$!
exec 3>"$W/a.in" 4>"$W/b.in"
N=0
# send <A|B> <sql...>  : send SQL then a marker; wait for marker output (max ~40s). Returns 1 on timeout (session dead/hung).
send() {
  local s=$1; shift
  local fd out
  if [ "$s" = A ]; then fd=3; out="$W/a.out"; else fd=4; out="$W/b.out"; fi
  N=$((N+1))
  printf '%s\n' "$*" >&$fd
  printf '\\echo MARK_%d_END\n' "$N" >&$fd
  local i
  for i in $(seq 1 800); do
    grep -q -x "MARK_${N}_END" "$out" 2>/dev/null && return 0
    sleep 0.05
  done
  return 1
}
# last_err <A|B> : print the last ERROR line emitted between the two most recent markers in THAT session's transcript
last_err() {
  local out; if [ "$1" = A ]; then out="$W/a.out"; else out="$W/b.out"; fi
  awk '/^MARK_[0-9]+_END$/{prev=cur; cur=NR} {line[NR]=$0} END{for(i=prev+1;i<cur;i++) if(line[i] ~ /^ERROR:/) buf=line[i]; print buf}' "$out"
}
alive() {  # B alive check: distinct query text, must round-trip
  send B "SELECT 'ALIVE_' || pg_backend_pid() AS alive;" || return 1
  grep -q '^ ALIVE_[0-9]' "$W/b.out" && ! tail -5 "$W/b.out" | grep -q 'server closed the connection unexpectedly\|connection to server was lost'
}

send A '\set VERBOSITY verbose' ; send B '\set VERBOSITY verbose'
send A '\pset pager off' ; send B '\pset pager off'
send A "SELECT version();"
send A "SELECT pg_backend_pid() AS a_pid;" ; send B "SELECT pg_backend_pid() AS b_pid;"
send B "SET yb_debug_log_catcache_events = on;"
send A "DROP TABLE IF EXISTS t; CREATE TABLE t (i int);"
log "SETUP a_pid=$(grep -A2 'a_pid' "$W/a.out" | grep -E '^ +[0-9]+' | head -1 | tr -d ' ') b_pid=$(grep -A2 'b_pid' "$W/b.out" | grep -E '^ +[0-9]+' | head -1 | tr -d ' ')"

classify() {  # classify an ERROR line
  case "$1" in
    *"Catalog Version Mismatch"*|*"catalog snapshot"*invalidated*|*MISMATCHED_SCHEMA*|*"schema version mismatch"*) echo MISMATCH_40001;;
    *"40001"*) echo CONFLICT_40001;;
    *"does not exist"*) echo STALE_RELCACHE_42703;;
    *"division by zero"*) echo DIVZERO_22012;;
    "") echo NONE;;
    *) echo OTHER;;
  esac
}

race_loop() {  # $1 = cell name, $2 = column prefix; runs the reporter loop until B errors; prints class
  local cell=$1 pfx=$2 n err cls
  send A "BEGIN ISOLATION LEVEL SERIALIZABLE;" ; send A "SELECT 1;"
  send B "BEGIN ISOLATION LEVEL SERIALIZABLE;" ; send B "SELECT 1;"
  cls=NONE
  for n in $(seq 1 30); do
    send A "ALTER TABLE t ADD COLUMN ${pfx}${n} int;"
    err=$(last_err A); [ -n "$err" ] && log "$cell A ALTER n=$n err=$err"
    send B "INSERT INTO t (${pfx}${n}) VALUES (1);" || { log "$cell B INSERT n=$n TIMEOUT"; cls=DEAD; break; }
    err=$(last_err B)
    if [ -n "$err" ]; then cls=$(classify "$err"); log "$cell B INSERT n=$n ERROR class=$cls :: $err"; break; fi
    log "$cell B INSERT n=$n ok"
  done
  send A "COMMIT;"
  echo "$cls"
}

# ---------------- R1: reporter literal, then COMMIT ----------------
cls=$(race_loop R1 c)
send B "COMMIT;" ; r1_commit=$(last_err B)
if alive; then r1_alive=ALIVE; else r1_alive=DEAD; fi
log "R1 RESULT trigger_class=$cls commit_err='${r1_commit}' b_after_commit=$r1_alive"

# ---------------- R2: reporter literal, then ROLLBACK ----------------
cls=$(race_loop R2 d)
send B "ROLLBACK;" ; r2_rb=$(last_err B)
if alive; then r2_alive=ALIVE; else r2_alive=DEAD; fi
log "R2 RESULT trigger_class=$cls rollback_err='${r2_rb}' b_after_rollback=$r2_alive"

# ---------------- R3: breaking DDL -> in-txn catalog version mismatch, then COMMIT ----------------
send B "BEGIN ISOLATION LEVEL SERIALIZABLE;" ; send B "SELECT count(*) FROM t;"
send A "ALTER TABLE t RENAME COLUMN i TO i0;" ; send A "ALTER TABLE t RENAME COLUMN i0 TO i;"
err=$(last_err A); [ -n "$err" ] && log "R3 A RENAME err=$err"
sleep 2.5   # let the heartbeat carry the breaking version to the tserver/shared memory
send B "INSERT INTO t (i) VALUES (1);" ; err=$(last_err B); cls3=$(classify "$err")
log "R3 B INSERT after breaking DDL class=$cls3 :: $err"
send B "COMMIT;" ; r3_commit=$(last_err B)
if alive; then r3_alive=ALIVE; else r3_alive=DEAD; fi
log "R3 RESULT trigger_class=$cls3 commit_err='${r3_commit}' b_after_commit=$r3_alive"

# ---------------- R4: deterministic aborted block + version bump + COMMIT ----------------
send B "BEGIN ISOLATION LEVEL SERIALIZABLE;" ; send B "SELECT 1;" ; send B "SELECT 1/0;"
err=$(last_err B); cls4=$(classify "$err"); log "R4 B forced error class=$cls4 :: $err"
send A "ALTER TABLE t ADD COLUMN bump1 int;" ; send A "ALTER TABLE t RENAME COLUMN bump1 TO bump2;"
sleep 3
send B "COMMIT;" ; r4_commit=$(last_err B)
if alive; then r4_alive=ALIVE; else r4_alive=DEAD; fi
log "R4 RESULT trigger_class=$cls4 commit_err='${r4_commit}' b_after_commit=$r4_alive"

# ---------------- C1: control, no concurrent DDL ----------------
send B "BEGIN ISOLATION LEVEL SERIALIZABLE;" ; send B "SELECT 1;" ; send B "INSERT INTO t (i) VALUES (42);"
err=$(last_err B); send B "COMMIT;" ; c1_commit=$(last_err B)
if alive; then c1_alive=ALIVE; else c1_alive=DEAD; fi
log "C1 RESULT insert_err='${err}' commit_err='${c1_commit}' b_after_commit=$c1_alive"

# ---------------- C2: control, aborted block, no DDL ----------------
send B "BEGIN ISOLATION LEVEL SERIALIZABLE;" ; send B "SELECT 1;" ; send B "SELECT 1/0;"
send B "COMMIT;" ; c2_commit=$(last_err B)
if alive; then c2_alive=ALIVE; else c2_alive=DEAD; fi
log "C2 RESULT commit_err='${c2_commit}' b_after_commit=$c2_alive"

send B "SELECT count(*) AS rows_in_t FROM t;"
send A '\q' ; send B '\q'
exec 3>&- 4>&-
wait $PA $PB 2>/dev/null

# ---------------- postgres / tserver log scan ----------------
PGLOG=$(ls -1 /tmp/ybd/data/yb-data/tserver/logs/postgresql-*.log 2>/dev/null | tail -1)
log "PGLOG=$PGLOG"
log "CRASH_SIGNATURES=$(grep -c -E 'terminated by signal|SIGSEGV|SIGABRT|TRAP:|PANIC|was terminated|server process .* exited|IsTransactionState' $PGLOG /tmp/ybd/data/yb-data/tserver/logs/yb-tserver.INFO 2>/dev/null | awk -F: '{s+=$NF} END{print s+0}')"
log "CLIENT_DISCONNECTS=$(grep -c -E 'server closed the connection unexpectedly|connection to server was lost' $W/a.out $W/b.out | awk -F: '{s+=$NF} END{print s+0}')"
log "PG_REFRESH_LINES=$(grep -i -c -E 'refreshing catalog cache|Preloading relcache complete|got tserver catalog version' $PGLOG 2>/dev/null || echo 0)"
BPID=$(grep -A2 'b_pid' "$W/b.out" | grep -E '^ +[0-9]+' | head -1 | tr -d ' ')
grep -i -E 'refreshing catalog cache|got tserver catalog version|Preloading relcache|invalidation messages|incremental' "$PGLOG" 2>/dev/null | grep -E "\[${BPID}\]" | tail -20 > "$W/pglog_tail.txt"
log "B_PID_REFRESH_EVENTS=$(wc -l < "$W/pglog_tail.txt" | tr -d ' ')"
echo "=== a.out ==="; cat "$W/a.out"; echo "=== b.out ==="; cat "$W/b.out"; echo "=== pglog_tail ==="; cat "$W/pglog_tail.txt"; echo "=== driver.log ==="; cat "$LOG"

**Comment by kriteshranjan** [comment]
AI validation notes (`yb5196-matrix.txt`):

# yb5196 runtime matrix evidence — 2026-09-10 (UTC). Host: podman arm64 VM, single-node yugabyted (RF1) per target, one container at a time.
# Cells: R1 reporter literal (SERIALIZABLE both; A ALTER ADD COLUMN cN / B INSERT (cN) until error; B COMMIT; B SELECT); R2 = R1 with ROLLBACK;
#        R3 breaking DDL (RENAME COLUMN there-and-back) vs open SERIALIZABLE txn, B INSERT (i) -> 40001, B COMMIT; R4 B SELECT 1/0 (aborted block), A ADD COLUMN+RENAME (version bump), sleep 3, B COMMIT;
#        C1 control txn INSERT COMMIT no DDL; C2 control aborted block COMMIT no DDL.  ALIVE = B answered a follow-up SELECT after COMMIT/ROLLBACK.
# CRASH_SIGNATURES = grep -c "terminated by signal|SIGSEGV|SIGABRT|TRAP:|PANIC|was terminated|server process .* exited|IsTransactionState" over postgresql-*.log + yb-tserver.INFO
# "got tserver catalog version" LOG lines (log_ysql_catalog_versions=true) are emitted by YBCheckSharedCatalogCacheVersion only when a refresh is needed -> they show WHEN B refreshed (always after COMMIT/ROLLBACK, never inside the aborted block).

===== target 2024.2.11.0-b36 =====
image digest: sha256:53c01c56315e245cc901cbc46e93f30fedb1f87833c7b1aa09d20fbeda27a288
PostgreSQL 11.2-YB-2024.2.11.0-b0 on aarch64-unknown-linux-gnu, compil
 Replication Factor: 1 
SETUP a_pid=405 b_pid=404
R1 B INSERT n=1 ok
R1 B INSERT n=2 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "c2" of relation "t" does not exist
R2 B INSERT n=1 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "d1" of relation "t" does not exist
R3 B INSERT after breaking DDL class=CONFLICT_40001 :: ERROR:  40001: current transaction is expired or aborted (query layer retry isn't possible because data was already sent, if this is the read com
R4 B forced error class=DIVZERO_22012 :: ERROR:  22012: division by zero
R1 RESULT trigger_class=STALE_RELCACHE_42703 commit_err='' b_after_commit=ALIVE
R2 RESULT trigger_class=STALE_RELCACHE_42703 rollback_err='' b_after_rollback=ALIVE
R3 RESULT trigger_class=CONFLICT_40001 commit_err='' b_after_commit=ALIVE
C1 RESULT insert_err='' commit_err='' b_after_commit=ALIVE
C2 RESULT commit_err='' b_after_commit=ALIVE
R4 RESULT trigger_class=DIVZERO_22012 commit_err='' b_after_commit=ALIVE
B_PID_REFRESH_EVENTS=5
CLIENT_DISCONNECTS=0
CRASH_SIGNATURES=0
-- crash scan matches (file:line:text) --
-- B backend catalog-version events in postgres log --
2026-09-10 16:05:20.686 UTC [404] LOG:  Refreshing catalog cache.
2026-09-10 16:05:23.741 UTC [404] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 6
2026-09-10 16:05:23.741 UTC [404] LOG:  Refreshing catalog cache.
2026-09-10 16:05:27.168 UTC [404] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 8
2026-09-10 16:05:27.168 UTC [404] LOG:  Refreshing catalog cache.
-- B transcript excerpt: COMMIT/ROLLBACK answers --
    1 COMMIT;	COMMIT
    4 COMMIT;	ROLLBACK
    1 ROLLBACK;	ROLLBACK

===== target 2025.1.4.0-b103 =====
image digest: sha256:08f1be0d6e074e0286e70067c12933a8cf536876d50fc43c52b1127d447ecdc0
PostgreSQL 15.12-YB-2025.1.4.0-b0 on aarch64-unknown-linux-gnu, compil
 Replication Factor: 1 
SETUP a_pid=424 b_pid=425
R1 B INSERT n=1 ok
R1 B INSERT n=2 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "c2" of relation "t" does not exist
R2 B INSERT n=1 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "d1" of relation "t" does not exist
R3 B INSERT after breaking DDL class=CONFLICT_40001 :: ERROR:  40001: current transaction is expired or aborted (query layer retry isn't possible because this is not the first command in the transacti
R4 B forced error class=DIVZERO_22012 :: ERROR:  22012: division by zero
R1 RESULT trigger_class=STALE_RELCACHE_42703 commit_err='' b_after_commit=ALIVE
R2 RESULT trigger_class=STALE_RELCACHE_42703 rollback_err='' b_after_rollback=ALIVE
R3 RESULT trigger_class=CONFLICT_40001 commit_err='' b_after_commit=ALIVE
C1 RESULT insert_err='' commit_err='' b_after_commit=ALIVE
R4 RESULT trigger_class=DIVZERO_22012 commit_err='' b_after_commit=ALIVE
C2 RESULT commit_err='' b_after_commit=ALIVE
B_PID_REFRESH_EVENTS=3
CLIENT_DISCONNECTS=0
CRASH_SIGNATURES=0
-- crash scan matches (file:line:text) --
-- B backend catalog-version events in postgres log --
2026-09-10 16:14:34.509 UTC [425] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 4
2026-09-10 16:14:37.371 UTC [425] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 6
2026-09-10 16:14:40.815 UTC [425] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 8
-- B transcript excerpt: COMMIT/ROLLBACK answers --
    1 COMMIT;	COMMIT
    4 COMMIT;	ROLLBACK
    1 ROLLBACK;	ROLLBACK

===== target 2025.2.6.0-b111 =====
image digest: sha256:4a1f7a94d3b70c61ecaf5e049521d691b16c9bf204ed01bdb6ab75d5beeb623e
PostgreSQL 15.12-YB-2025.2.6.0-b0 on aarch64-unknown-linux-gnu, compil
 Replication Factor: 1 
SETUP a_pid=454 b_pid=455
R1 B INSERT n=1 ok
R1 B INSERT n=2 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "c2" of relation "t" does not exist
R2 B INSERT n=1 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "d1" of relation "t" does not exist
R3 B INSERT after breaking DDL class=CONFLICT_40001 :: ERROR:  40001: current transaction is expired or aborted (query layer retry isn't possible because this is not the first command in the transacti
R4 B forced error class=DIVZERO_22012 :: ERROR:  22012: division by zero
R1 RESULT trigger_class=STALE_RELCACHE_42703 commit_err='' b_after_commit=ALIVE
R2 RESULT trigger_class=STALE_RELCACHE_42703 rollback_err='' b_after_rollback=ALIVE
R3 RESULT trigger_class=CONFLICT_40001 commit_err='' b_after_commit=ALIVE
C1 RESULT insert_err='' commit_err='' b_after_commit=ALIVE
C2 RESULT commit_err='' b_after_commit=ALIVE
R4 RESULT trigger_class=DIVZERO_22012 commit_err='' b_after_commit=ALIVE
B_PID_REFRESH_EVENTS=2
CLIENT_DISCONNECTS=0
CRASH_SIGNATURES=0
-- crash scan matches (file:line:text) --
-- B backend catalog-version events in postgres log --
2026-09-10 15:52:59.179 UTC [455] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 6
2026-09-10 15:53:02.539 UTC [455] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 8
-- B transcript excerpt: COMMIT/ROLLBACK answers --
    1 COMMIT;	COMMIT
    4 COMMIT;	ROLLBACK
    1 ROLLBACK;	ROLLBACK

===== target 2026.1.1.1-b2 =====
image digest: sha256:d27eede72bbc669e348e9920318589b4bb68e98204a241ec6bfaba04bf4f6699
PostgreSQL 15.12-YB-2026.1.1.1-b0 on aarch64-unknown-linux-gnu, compil
 Replication Factor: 1 
SETUP a_pid=524 b_pid=523
R1 B INSERT n=1 ok
R1 B INSERT n=2 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "c2" of relation "t" does not exist
R2 B INSERT n=1 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "d1" of relation "t" does not exist
R3 B INSERT after breaking DDL class=CONFLICT_40001 :: ERROR:  40001: current transaction is expired or aborted (query layer retry isn't possible because this is not the first command in the transacti
R4 B forced error class=DIVZERO_22012 :: ERROR:  22012: division by zero
R1 RESULT trigger_class=STALE_RELCACHE_42703 commit_err='' b_after_commit=ALIVE
R2 RESULT trigger_class=STALE_RELCACHE_42703 rollback_err='' b_after_rollback=ALIVE
R3 RESULT trigger_class=CONFLICT_40001 commit_err='' b_after_commit=ALIVE
C1 RESULT insert_err='' commit_err='' b_after_commit=ALIVE
C2 RESULT commit_err='' b_after_commit=ALIVE
R4 RESULT trigger_class=DIVZERO_22012 commit_err='' b_after_commit=ALIVE
CLIENT_DISCONNECTS=0
CRASH_SIGNATURES=1
B_PID_REFRESH_EVENTS=4
-- crash scan matches (file:line:text) --
/tmp/ybd/data/yb-data/tserver/logs/postgresql-2026-09-10_154944.log:14:2026-09-10 15:49:44.488 UTC [279] WARNING:  server process (PID 286) exited with exit code 1
-- B backend catalog-version events in postgres log --
2026-09-10 15:49:47.173 UTC [523] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 2
2026-09-10 15:49:47.864 UTC [523] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 5
2026-09-10 15:49:50.723 UTC [523] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 7
2026-09-10 15:49:54.110 UTC [523] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 9
-- B transcript excerpt: COMMIT/ROLLBACK answers --
    1 COMMIT;	COMMIT
    4 COMMIT;	ROLLBACK
    1 ROLLBACK;	ROLLBACK

===== target 2.25.2.0-b359 =====
image digest: sha256:3f9cd1b8f520cc78e82ace51174be73343a551148bc6c9a9f821a22f1d2858c5
PostgreSQL 15.12-YB-2.25.2.0-b0 on aarch64-unknown-linux-gnu, compiled
 Replication Factor: 1 
SETUP a_pid=397 b_pid=396
R1 B INSERT n=1 ok
R1 B INSERT n=2 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "c2" of relation "t" does not exist
R2 B INSERT n=1 ERROR class=STALE_RELCACHE_42703 :: ERROR:  42703: column "d1" of relation "t" does not exist
R3 B INSERT after breaking DDL class=CONFLICT_40001 :: ERROR:  40001: could not serialize access due to concurrent update (query layer retry isn't possible because this is not the first command in the
R4 B forced error class=DIVZERO_22012 :: ERROR:  22012: division by zero
R1 RESULT trigger_class=STALE_RELCACHE_42703 commit_err='' b_after_commit=ALIVE
R2 RESULT trigger_class=STALE_RELCACHE_42703 rollback_err='' b_after_rollback=ALIVE
R3 RESULT trigger_class=CONFLICT_40001 commit_err='' b_after_commit=ALIVE
R4 RESULT trigger_class=DIVZERO_22012 commit_err='' b_after_commit=ALIVE
C1 RESULT insert_err='' commit_err='' b_after_commit=ALIVE
C2 RESULT commit_err='' b_after_commit=ALIVE
B_PID_REFRESH_EVENTS=2
CLIENT_DISCONNECTS=0
CRASH_SIGNATURES=0
-- crash scan matches (file:line:text) --
-- B backend catalog-version events in postgres log --
2026-09-10 16:17:57.508 UTC [396] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 6
2026-09-10 16:18:00.957 UTC [396] LOG:  YBCheckSharedCatalogCacheVersion: got tserver catalog version: 8
-- B transcript excerpt: COMMIT/ROLLBACK answers --
    1 COMMIT;	COMMIT
    4 COMMIT;	ROLLBACK
    1 ROLLBACK;	ROLLBACK

# Note: 2026.1.1.1-b2 run 2 shows one "server process (PID 286) exited with exit code 1" WARNING at postgres-log line 14 (15:49:44), ~3 s before sessions A/B (PIDs 522/523) connected; run 1 of the same image had 0 matches. Startup-time exit code 1 (not a signal), unrelated to the scenario.

**Comment by kriteshranjan** [comment]
Closing this issue based on AI validation. cc: @m-iancu , @VamsidharGunturu
